src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+    if (this.health <= 0) {
+      this.health = 0;
+      this.remove();
+    }
+  }
+
+  remove() {
+    const index = Animal.alive.indexOf(this);
+    Animal.alive.splice(index, 1);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  decreaseHealth(amount = 50) {
+    super.decreaseHealth(amount);
+  }
```

Comment:
The `decreaseHealth` method in the `Herbivore` class does not need to be overridden if it only calls the parent method with a default parameter. You can rely on the `decreaseHealth` method from the `Animal` class directly unless there is a specific reason for this override.